lib/info-extras.js

```diff
@@ -124,11 +124,49 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', ',\n');
+  let watchNextJson;
   try {
     jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
+    watchNextJson = JSON.parse(jsonStr.watch_next_response);
+  }
+  catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  let rvsParams = jsonStr.rvs.split(',').map((e) => qs.parse(e));
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results;
+  let videos = [];
+  for (let result of secondaryResults) {
+    let details = result.compactVideoRenderer;
+    if (details) {
+      try {
+        let viewCount = details.viewCountText.simpleText;
+        if (!viewCount) {
+          viewCount = details.viewCountText.runs[0].text;
+        }
+        let shortViewCount = details.shortViewCountText.simpleText;
+        let shortViewCountTest = /^\d/.test(shortViewCount);
+        if (!shortViewCountTest) {
+          let shortViewCountFromRVS = rvsParams.find((elem) => elem.id === details.videoId)
+          if (shortViewCountFromRVS) {
+            shortViewCount = shortViewCountFromRVS;
+          }
+        }
+        viewCount = (/^\d/.test(viewCount) ? viewCount : shortViewCount).split(' ')[0]; 
+        let lengthSeconds = parseTime.humanStr(details.lengthText.simpleText);
```

Comment:
not sure how this is working here but, `parseTime` should be `require()`d at the top of the file

Reply:
would also prefer if this call to `humanStr()` was placed inline in the line below, since `lengthSeconds` is only used there

Reply:
> not sure how this is working here but, `parseTime` should be `require()`d at the top of the file
sorry it wasn't working I was trying with another file, added the import

Reply:
> would also prefer if this call to `humanStr()` was placed inline in the line below, since `lengthSeconds` is only used there

👍